### PR TITLE
Add enum have flag

### PR DESCRIPTION
**Pull request by bangsholt**
As noted in PR #272, I've added those methods.

I'm currently unable to figure out a way to use reflection in the portable library to get the attributes on the enum class.

### CODE REVIEW

File: src/Shouldly.Shared/ShouldlyExtensionMethods/ShouldBeEnumExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using JetBrains.Annotations;
+
+namespace Shouldly.ShouldlyExtensionMethods
+{
+    public static class ShouldHaveEnumExtensions
+    {
+        public static void ShouldHaveFlag(this Enum actual, Enum expectedFlag) 
+            => ShouldHaveFlag(actual, expectedFlag, () => null);
```
Comment (JakeGinnivan) [inline src/Shouldly.Shared/ShouldlyExtensionMethods/ShouldBeEnumExtensions.cs]: Awesome! Love the use of the method expression syntax :)

File: src/Shouldly.Shared/ShouldlyExtensionMethods/ShouldBeEnumExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
+﻿#define PORTABLE
```
Comment (JakeGinnivan) [inline src/Shouldly.Shared/ShouldlyExtensionMethods/ShouldBeEnumExtensions.cs]: You shouldn't need the define here?

Comment (bangsholt) [inline src/Shouldly.Shared/ShouldlyExtensionMethods/ShouldBeEnumExtensions.cs]: I don't know how to switch profiles, that's why I did the define.

I'm sure there's an easy way, can you enlighten me? :)

**Comment by bangsholt** [comment]
I've merged in your changes, but I would like to add in docs either today or tomorrow before it's merged.

**Comment by bangsholt** [comment]
I'm still on it :)

**Comment by JakeGinnivan** [comment]
Howd you get on with the docs?

**Comment by bangsholt** [comment]
I've played with it, but I seem to get an exception that my diff tool can't be located.

**Comment by JosephWoodward** [comment]
I've created the docs for these changes in [this branch](https://github.com/JosephWoodward/shouldly/tree/AddEnumHaveFlag). What's the best way to push them to this pull request, or does git require a a different approach?

**Comment by JakeGinnivan** [comment]
You can do a PR to @bangsholt's branch in his fork. Otherwise I can merge this and docs come as a separate PR.

@bangsholt what mergetool do you use?

**Comment by bangsholt** [comment]
I just realized I haven't installed Beyond Compare on my dev VM.

But I also noticed after digging, that the files were actually being generated, but since @JosephWoodward has kindly contributed the docs, it makes more sense to use his as I was stuck on the exception.

**Comment by JosephWoodward** [comment]
Thanks @JakeGinnivan I'll create a PR for you tonight @bangsholt so we can get this one closed off :)

**Comment by JosephWoodward** [comment]
PR created @bangsholt (https://github.com/bangsholt/shouldly/pull/2)

**Comment by JosephWoodward** [comment]
I've manually merged these and moved them to #361.

**Pull request closed without merging**